src/plugins/workspace/server/saved_objects/workspace_id_consumer_wrapper.ts

```diff
@@ -48,25 +49,70 @@ export class WorkspaceIdConsumerWrapper {
     return type === UI_SETTINGS_SAVED_OBJECTS_TYPE;
   }
 
+  private async checkWorkspacesExist(
+    finalOptions: SavedObjectsCreateOptions,
+    wrapperOptions: SavedObjectsClientWrapperOptions
+  ) {
+    if (finalOptions.workspaces?.length) {
+      let isAllTargetWorkspaceExisting = false;
+      // If only has one workspace, we should use get to optimize performance
+      if (finalOptions.workspaces.length === 1) {
+        const workspaceGet = await this.workspaceClient.get(
+          { request: wrapperOptions.request },
+          finalOptions.workspaces[0]
+        );
+        if (workspaceGet.success) {
+          isAllTargetWorkspaceExisting = true;
+        }
+      } else {
+        const workspaceList = await this.workspaceClient.list(
+          {
+            request: wrapperOptions.request,
+          },
+          {
+            perPage: 9999,
+          }
+        );
+        if (workspaceList.success) {
+          const workspaceIdsSet = new Set(
+            workspaceList.result.workspaces.map((workspace) => workspace.id)
+          );
+          isAllTargetWorkspaceExisting = finalOptions.workspaces.every((targetWorkspace) =>
+            workspaceIdsSet.has(targetWorkspace)
+          );
+        }
+      }
+
+      if (!isAllTargetWorkspaceExisting) {
+        throw SavedObjectsErrorHelpers.decorateBadRequestError(
+          new Error(
+            i18n.translate('workspace.id_consumer.invalid', {
+              defaultMessage: 'Invalid workspaces',
+            })
+          )
+        );
+      }
+    }
+  }
+
   public wrapperFactory: SavedObjectsClientWrapperFactory = (wrapperOptions) => {
     return {
       ...wrapperOptions.client,
-      create: <T>(type: string, attributes: T, options: SavedObjectsCreateOptions = {}) =>
-        wrapperOptions.client.create(
-          type,
-          attributes,
-          this.isConfigType(type)
-            ? options
-            : this.formatWorkspaceIdParams(wrapperOptions.request, options)
-        ),
-      bulkCreate: <T = unknown>(
+      create: async <T>(type: string, attributes: T, options: SavedObjectsCreateOptions = {}) => {
```

Comment:
Shall we do the check for addToWorkspace?

Reply:
Do you think we also need to check the update and bulk update api?

Reply:
I think it should be fine for update and bulkUpdate because these 2 APIs won't touch `workspaces` fields.

Reply:
We can not check addToWorkspace in case of Associate/UnAssociate method will not call id_consumer_wrapper.